codice/ddf
# DDF-4217: SolrCatalogProvider does not respect query filters when querying by exact match to id

#### What does this PR do?
Adds filter query capability to Solr real time queries.

#### Who is reviewing it? 
@peterhuffer 
@tbatie 
@ethantmanns 

#### Select relevant component teams: 
@codice/solr 

#### Ask 2 committers to review/merge the PR and tag them here.

@clockard
@pklinef

#### How should this be tested?
1) Unzip ddf
2) Change solr to http instead of https in etc/custom.system.properties
3)  For Windows set:
set SOLR_OPTS=%SOLR_OPTS% -Dsolr.autoSoftCommit.maxTime=120000
set SOLR_OPTS=%SOLR_OPTS% -Dsolr.autoCommit.maxTime=180000
in ddf-2.14.0-SNAPSHOT\solr\bin\solr.in.cmd
For *nix set:
SOLR_OPTS=$SOLR_OPTS -Dsolr.autoSoftCommit.maxTime=120000
SOLR_OPTS=$SOLR_OPTS -Dsolr.autoCommit.maxTime=180000
in ddf-2.14.0-SNAPSHOT\solr\bin\solr.in.sh
4) Start DDF
5) Set logging
log:set DEBUG ddf.catalog.source.solr
6) Ingest a metacard via the REST endpoint using curl:
eample>$ curl -k -X POST -i -H "Content-Type: multipart/mixed" -F parse.metadata=@./metacard-1.xml https://localhost:8993/services/catalog
You will get the metacard ID back in the response
7) Verify that the metacard is not visible via the Solr Admin UI:
http://localhost:8994/solr
q: metacard-tags:txt:resource should show 0 results
8) Query via Intrigue by ID=<metacard-id> (real time query) and title=<metacard title> and verify that you get the metacard back

#### Any background context you want to provide?

#### What are the relevant tickets?
https://codice.atlassian.net/browse/DDF-4217
#### Screenshots

#### Checklist:

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.
